src/Crdt.Tests/ResourceTests/RemoteResourcesTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Runtime.CompilerServices;
+using Crdt.Resource;
+using FluentAssertions.Formatting;
+
+namespace Crdt.Tests.ResourceTests;
+
+public class RemoteResourcesTests : DataModelTestBase
+{
+    private RemoteServiceMock _remoteServiceMock = new();
+
+    public RemoteResourcesTests()
+    {
+    }
+
+    private string CreateFile(string contents, [CallerMemberName] string fileName = "")
+    {
+        var filePath = Path.GetFullPath(fileName + ".txt");
+        File.WriteAllText(filePath, contents);
+        return filePath;
+    }
+
+    private async Task<(Guid resourceId, string remoteId)> SetupRemoteResource(string fileContents)
+    {
+        var remoteId = _remoteServiceMock.CreateRemoteResource(fileContents);
+        var resourceId = Guid.NewGuid();
+        await DataModel.AddChange(_localClientId, new CreateRemoteResourceChange(resourceId, remoteId));
+        return (resourceId, remoteId);
+    }
+
+    private async Task<(Guid resourceId, string localPath)> SetupLocalFile(string contents, [CallerMemberName] string fileName = "")
+    {
+        var file = CreateFile(contents, fileName);
+        //because resource service is null the file is not uploaded
+        var resourceId = await DataModel.AddLocalResource(file, _localClientId, resourceService: null);
+        return (resourceId, file);
+    }
+
+    [Fact]
+    public async Task CreatingAResourceResultsInPendingLocalResources()
+    {
+        var (_, file) = await SetupLocalFile("contents");
+
+        //act
+        var pending = await DataModel.ListResourcesPendingUpload();
+
+
+        pending.Should().ContainSingle().Which.LocalPath.Should().Be(file);
+    }
+
+    [Fact]
+    public async Task ResourcesNotLocalShouldShowUpAsNotDownloaded()
+    {
+        var (resourceId, remoteId) = await SetupRemoteResource("test");
+
+        //act
+        var pending = await DataModel.ListResourcesPendingDownload();
+
+
+        var remoteResource = pending.Should().ContainSingle().Subject;
+        remoteResource.RemoteId.Should().Be(remoteId);
+        remoteResource.Id.Should().Be(resourceId);
+    }
+
+    [Fact]
+    public async Task CanUploadFileToRemote()
+    {
+        var fileContents = "resource";
+        var localFile = CreateFile(fileContents);
+
+        //act
+        var resourceId =
+            await DataModel.AddLocalResource(localFile, _localClientId, resourceService: _remoteServiceMock);
+
+
+        var resource = await DataModel.GetLatest<RemoteResource>(resourceId);
```

Comment:
Shouldn't the //act go here?

Reply:
No this is the assert. It's just fetching what we put in the database on the add local call.